master-docker-nonstandard/master.cfg

```diff
@@ -793,10 +793,21 @@ f_valgrind_build.addStep(getSourceTarball())
 f_valgrind_build.addStep(
     steps.Compile(
         command=[
-            "sh",
+            "bash",
             "-c",
             util.Interpolate(
-                "cmake . -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DENABLE_ASSEMBLER=1 -DWITH_EXTRA_CHARSETS=complex -DENABLE_THREAD_SAFE_CLIENT=1 -DWITH_BIG_TABLES=1 -DWITH_PLUGIN_ARIA=1 -DWITH_ARIA_TMP_TABLES=1 -DWITH_JEMALLOC=NO -DCMAKE_BUILD_TYPE=Debug -DSECURITY_HARDENED=OFF -DWITH_VALGRIND=1 -DHAVE_LIBAIO_H=0 -DCMAKE_DISABLE_FIND_PACKAGE_URING=1 -DCMAKE_DISABLE_FIND_PACKAGE_LIBAIO=1 -DWITH_SSL=bundled -DWITH_MAX=AUTO -DWITH_EMBEDDED_SERVER=1 -DWITH_LIBEVENT=bundled -DPLUGIN_PLUGIN_FILE_KEY_MANAGEMENT=NO -DPLUGIN_TEST_SQL_DISCOVERY=DYNAMIC -DPLUGIN_TOKUDB=NO -DPLUGIN_ROCKSDB=NO -DPLUGIN_AUTH_GSSAPI=NO -DENABLE_LOCAL_INFILE=1 -DMYSQL_SERVER_SUFFIX=-valgrind-max -DWITH_DBUG_TRACE=OFF && make -j%(kw:jobs)s",
+                """cmake . \\
+                -DCMAKE_C_COMPILER_LAUNCHER=ccache \\
+                -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \\
+                -DCMAKE_BUILD_TYPE=Debug \\
+                -DWITH_VALGRIND=1 \\
+                -DWITH_MAX=1 \\
+                -DWITH_EMBEDDED_SERVER=1 \\
+                -DWITH_DBUG_TRACE=OFF \\
+                -DPLUGIN_TOKUDB=NO \\
+                -DEXTRA_DEBUG=1 \\
+                -DSECURITY_HARDENED=OFF \\
+                && make -j%(kw:jobs)s""",
```

Comment:
Tested on Fedora 40, valgrind-3.23.0, with g{cc,++} = 14.1.1

Need one Archive storage engine patch to compile with gcc, and one innodb patch to stop a valgrind hit on a two assertions. Will submit Monday.

Tests that timed out for me and would need not-valgrind exclusions in the server codebase:
* perfschema.statement_program_non_nested
* perfschema.statement_program_nesting_event_check

WITH_MAX - not needed - the auto-detection is fine.

WITH_DBUG_TRACE_OFF - good.

EXTRA_DEBUG=1, probably ok.

Didn't need to run SECURITY_HARDENED=OFF.

As we don't run `mtr --embedded`, compiling the EMBEDDED_SERVER is a waste. Lets omit.

mtr run:
* remove `^perfschema\.short_option_1$` - it was a quick and succesful test.
* encryption test are slow which is probably why its here, most seem to pass, but 4 mins each (out of 116 tests).
* --mysqld="--loose-innodb-purge-threads=1" - not required. Passes without this and since 4 is default, detecting confict/race conditions is good.

Reply:
Just re-pushed with the latest info, thank you for the suggestions.

I was testing locally on the current Valgrid image, 
and it seems like setting debug off it will skip a lot of tests with the message "Requires Debug build".

Is this ok, it won't decrease Valgrind's coverage or it's a better to run tests as in a server production like build configuration?

As for bumping to a new Valgrind version by using Noble/Fedora40, can we push these changes for the current image and then come up with a builder upgrade? The CI for the current Valgrind image is missing from the buildbot configuration and I should look closely how to reproduce the original Dockerfile in the new fashion of building the complete docker file from multiple files.

Reply:
Yes, there are more debug mode tests.

These are tests that start/stop threads at different points to get repeatable results. For the most part the same execution path occurs without a debug build and most debug tests. For the testing of a memory leak the execution. The absence of control points adds a bit more entropy so might catch new errors. There may be a couple of code paths less tested. By for the most, its testing something closer to server production builds.

An older valgrind might depend bit more on the original sse cflags for the strcmp. Older gcc versions are probably keeping of those feature being exposed. Sure bump for now. The valgrind should be a basic package add to the base image and reusing that.  Then just the buildbot changes to to use that image.